game-core/src/main/java/games/strategy/engine/framework/startup/ui/FileBackedGamePropertiesCache.java

```diff
@@ -68,10 +68,10 @@ public void loadCachedGamePropertiesInto(final GameData gameData) {
         try (InputStream is = new FileInputStream(cache);
             ObjectInputStream in = new ObjectInputStream(is)) {
           final Map<String, Serializable> serializedMap = (Map<String, Serializable>) in.readObject();
-          for (final IEditableProperty property : gameData.getProperties().getEditableProperties()) {
+          for (final IEditableProperty<?> property : gameData.getProperties().getEditableProperties()) {
             final Serializable ser = serializedMap.get(property.getName());
             if (ser != null) {
-              property.setValue(ser);
+              property.setValueIfValid(ser);
```

Comment:
The additional validation here is probably the right thing to do.  However, could we now silently discard a value?  Would logging the discarded value be appropriate?  If so, should it just be done by the caller (possibly using a `boolean` return value from `setValueIfValid()`), or should it be internal to `setValueIfValid()`?